app/views/bookings/show.html.erb

```diff
@@ -125,6 +125,20 @@
                             <div>Total (A+B+C+D) = <%= @booking.cost %> </div>
                         </td>
                     </tr>
+                    <% elsif @booking.is_cancelled? %>
+                    <tr>
+                        <th>
+                            Cost Calculations
+                        </th>
+                        <td>
+                             <% if 0.5.day.from_now > DateTime.parse(@booking.start_date.to_s + "T" + @booking.time_in)%>
```

Comment:
You can do something like... if booking.cost == @per_day cost then its full charges if its == 0.5 * per_day then its 1/2 day charges.